app/channel.go

```diff
@@ -2408,8 +2408,15 @@ func (a *App) MarkChannelAsUnreadFromPost(postID string, userID string, collapse
 		threadMembership, _ := a.Srv().Store.Thread().GetMembershipForUser(user.Id, threadId)
 		// if this post was not followed before, create thread membership and update mention count
 		if threadMembership == nil {
-			threadMembership, _ = a.Srv().Store.Thread().MaintainMembership(user.Id, threadId, true, true, true, true, false)
-			threadData, _ := a.Srv().Store.Thread().Get(threadId)
+			opts := store.ThreadMembershipOpts{
+				Following:             true,
+				IncrementMentions:     true,
+				UpdateFollowing:       true,
+				UpdateViewedTimestamp: true,
+				UpdateParticipants:    false,
+			}
+			threadMembership, _ = a.Srv().Store.Thread().MaintainMembership(context.Background(), user.Id, threadId, opts)
+			threadData, _ := a.Srv().Store.Thread().Get(context.Background(), threadId)
```

Comment:
What was the reason these errors are assigned the blank identifier (I realize it was there before your change)?

Reply:
I can't really answer that and I believe the author is no longer working with us so I can only speculate. To me it looks like we are trying to avoid handling the errors and simply check if they return non-nil to proceed.

Reply:
I believe the author is not working with us anymore so I can only speculate. It looks to me like we are just avoiding error handling and only proceed if the returned object are not `nil`.